File: doc/Programs/LecturePrograms/programs/IntroProgramming/Fortran/program2.f90
PROGRAM binary_integer
IMPLICIT NONE
  INTEGER  i, number, terms(32) ! storage of a0, a1, etc, up to 32 bits
  
  WRITE(*,*) 'Give a number to transform to binary notation' 
  READ(*,*) number
! Initialise the terms a0, a1 etc
  terms = 0
! Fortran takes only integer loop variables
  DO i=0, 31
     terms(i) = MOD(number,2)
     number = number/2
  ENDDO
! write out results
  WRITE(*,*) 'Binary representation '
  DO i=0, 31
    WRITE(*,*)' Term nr and value', i, terms(i)
  ENDDO

END PROGRAM binary_integer

File: doc/Programs/LecturePrograms/programs/IntroProgramming/cpp/program2.cpp
#include <iostream>
#include <cmath>
#include <cstdio>
#include <cstdlib>
#include <iostream>

using namespace std;

int main (int argc, char* argv[])
{
  int i; 
  int terms[32]; // storage of a0, a1, etc, up to 32 bits
  int number = atoi(argv[1]); 
  // initialise the term a0, a1 etc
  for (i=0; i < 32 ; i++){ terms[i] = 0;}
  for (i=0; i < 32 ; i++){ 
    terms[i] = number%2;
    number /= 2;
  }
  // write out results
  cout << "Number of bytes used= " << sizeof(number) << endl;
  for (i=0; i < 32 ; i++){ 
    cout << " Term nr: " << i << "Value= " << terms[i];
    cout << endl;
  }
  return 0;  
}
